cave/plot/configurator_footprint.py

```diff
@@ -355,131 +371,174 @@ def get_conf_matrix(self):
         return np.array(conf_matrix), conf_list, runs_runs_conf
 
     def _get_size(self, r_p_c):
-        return 10 + ((r_p_c - self.min_runs_per_conf) / (self.max_runs_per_conf - self.min_runs_per_conf)) * 40
-
-    def plot(self, X, conf_list: list, runs_runs_conf, inc_list: list,
-             runs_labels=None, contour_data=None):
-        '''
-            plots sampled configuration in 2d-space;
-            saves results in self.output, if set
+        sizes = 10 + ((r_p_c - self.min_runs_per_conf) / (self.max_runs_per_conf - self.min_runs_per_conf)) * 40
+        sizes *= np.array([0 if r == 0 else 1 for r in r_p_c])  # 0 size if 0 runs
```

Comment:
Is the multiplication necessary? I'd assume that Configurations with 0 runs would never appear in the plot anyways

Reply:
The thing is, that our `conf_list` holds all configs that were evaluated in any of the passed runhistories. I think it's very good, because like this we can extend the footprint without refactoring if need be. On the downside we might have have configuration-entries with 0 ta-runs for some of the configurator-runs, because the number of ta-runs is per configurator-run (so for n configurator runs (=runhistories), there are n lists that have entries for all the configurations). While I think we should keep that structure for simplicity, it makes this specific multiplication necessary for this specific strategy of minimum sizes... :)